### PR TITLE
Add ability to subscribe to multicast channels

### PR DESCRIPTION
**Pull request by poulh**
With this change I can subscribe to multicast data by mcastjoining the passed in host/port.

before i could not access the socket to do the mcast subscription.

### CODE REVIEW

**Comment by sodabrew** [comment]
Thanks! Reviewing today!

File: lib/em/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module EventMachine
-  VERSION = "1.2.0.1"
+  VERSION = "1.2.0.2"
```
Comment (sodabrew) [inline lib/em/version.rb]: Please don't bump the version in a PR - I'll do that when I release a group of changes.

File: lib/eventmachine.rb [hunk 2]
```diff
@@ -881,6 +881,19 @@ def self.open_datagram_socket address, port, handler=nil, *args
     c
   end
 
+  # Same as open_datagram_socket but also multicast subscribes
+  def self.open_datagram_socket_and_mcastjoin address, port, handler=nil, *args
+
+    klass = klass_from_handler(Connection, handler, *args)
+    s = EM::open_udp_socket address, port.to_i
```
Comment (sodabrew) [inline lib/eventmachine.rb]: You don't need the `EM::` namespace since you're already inside `EM`

**Comment by poulh** [comment]
Hi,

It turns out you can do this without my commit.  I didn't realize it until after I sent the pull request.  The following code works with the existing open_datagram_socket method
```

 udp_handler = EM::open_datagram_socket(host, port, MulticastHandler)
 ip =  IPAddr.new( host ).hton + IPAddr.new("0.0.0.0").hton
 udp_handler.set_sock_opt Socket::IPPROTO_IP, Socket::IP_ADD_MEMBERSHIP, ip
```

Maybe its worth putting this in an example?  I found it difficult to find the answer googling around.

File: lib/eventmachine.rb [hunk 3]
```diff
@@ -881,6 +881,19 @@ def self.open_datagram_socket address, port, handler=nil, *args
...
+    ip =  IPAddr.new( address ).hton + IPAddr.new("0.0.0.0").hton
```
Comment (sodabrew) [inline lib/eventmachine.rb]: I don't remember enough about multicast if the interface address should be configurable here or just use the any address?

File: lib/eventmachine.rb [hunk 4]
```diff
@@ -881,6 +881,19 @@ def self.open_datagram_socket address, port, handler=nil, *args
...
+    set_sock_opt s, Socket::IPPROTO_IP, Socket::IP_ADD_MEMBERSHIP, ip
```
Comment (sodabrew) [inline lib/eventmachine.rb]: I'm a little worried about this creating a precedent for adding separate top-level methods for each sockopt you might want to set. Could you do the IP_ADD_MEMBERSHIP in the post_init of your connection handler class?

Comment (sodabrew) [inline lib/eventmachine.rb]: Poke @poulh

**Comment by sodabrew** [comment]
Per comment https://github.com/eventmachine/eventmachine/pull/736#issuecomment-251477301 this PR isn't required because you can call set_sock_opt on the underlying socket with arguments to join a multicast channel.

**Pull request closed without merging**